src/modeling_library/distributions/dirichlet.jl

```diff
@@ -0,0 +1,42 @@
+struct Dirichlet <: Distribution{Float64} end
+
+"""
+    Dirichlet(alpha::Vector{Float64})
+
+Sample a `Float64` from a Dirichlet distribution.
+"""
+const dirichlet = Dirichlet()
+
+function logpdf(::Dirichlet, x::AbstractVector{T}, alpha::AbstractVector{U}) where {T <: Real, U <: Real}
+    if isapprox(sum(x), 1) && all(x .>= 0) && all(alpha .>= 0)
+        ll = sum((a_i - 1) * log(x_i) for (a_i, x_i) in zip(alpha, x))
+        ll -= sum(loggamma.(alpha)) - loggamma(sum(alpha))
+        ll
+    else
+        -Inf
+    end
+end
+
+function logpdf_grad(::Dirichlet, x::AbstractVector{T}, alpha::AbstractVector{U}) where {T <: Real, U <: Real}
+    if isapprox(sum(x), 1) && all(x .>= 0) && all(alpha .>= 0)
+        deriv_x = (alpha .- 1) ./ x
+        deriv_alpha = log.(x) .- digamma.(alpha) .+ digamma(sum(alpha))
+        (deriv_x, deriv_alpha)
+    else
+        (0., 0.)
```

Comment:
Shouldn't these return values have the same shape as the input vectors, even if they are all zero? i.e. the gradient with respect to `x` should have length of `x`, and gradient with respect to `alpha` should have length `alpha`.

Maybe also add a check to both `logpdf` and `logpdf_grad` requiring that both `x` and `alpha` have the same length.

Reply:
Ah, good catch